astroid/scoped_nodes.py

```diff
@@ -105,7 +105,12 @@ def clean_duplicates_mro(sequences, cls, context):
             (node.lineno, node.qname()) if node.name else None for node in sequence
         ]
         last_index = dict(map(reversed, enumerate(names)))
-        if names and names[0] is not None and last_index[names[0]] != 0:
+        if (
+            names
+            and names[0] is not None
+            and last_index[names[0]] != 0
+            and names[0][1] != "typing._GenericAlias"
+        ):
```

Comment:
I'm not found of this kind of solution. I think there is maybe something to handle through TypingBrain instead.
However i'm not sure, i'm still investigating this.
By the way on my machine the bug is also present with python 3.9.

Reply:
I understand. Thanks for your work!

> By the way on my machine the bug is also present with python 3.9.

Hmm, interesting. MacOS with Py 3.9.2 works. Please let me know if I can help test things, once you have found a better solution.